theano/gpuarray/opt.py

```diff
@@ -381,6 +380,10 @@ def apply(self, fgraph):
             else:
                 outputs = new_ops(*[mapping[i] for i in node.inputs],
                                   return_list=True)
+
+            if new_ops:
+                node_created[lopt] += len(theano.gof.graph.ops([mapping[i] for i in node.inputs], node.outputs))
```

Comment:
Do this change:

node.outputs -> outputs

This is an important change. Otherwise, it is mixing the new and old graph, so would traverse the full old graph each time.